python-frontend/typeshed_serializer/serializer/serializers.py

```diff
@@ -94,6 +102,15 @@ class Serializer(ABC):
     save_location: str
     output_folder: str
 
+    logger = logging.getLogger(__name__)
```

Comment:
Is the logging needed as part of the `scikit-learn` stubs change?

I'm not against having it, but I think this (as well as the formatting change) would be nice to have in a separate commit to clearly separate the QoL improvements for the serializer from the functional changes.

Also, I think if we have logs, we should have assertions for those logs in our tests (it's actually a point I like to make when referring to our string formatting rules, people tend not to make assertions on logs and we did find a few interesting issues in some big projects due to that).

Reply:
No, you are right this should have been in a different commit. For the sake of time, I will merge this PR. But I will better organize the commit history in the following PRs